psi4/driver/procrouting/dft_funcs/dict_builder.py

```diff
@@ -0,0 +1,207 @@
+    ):
+        raise ValidationError(
+            "SCF: Libxc parameters requested for an exchange functional not defined as a component of functional %s." %
+            (name))
+
+
+def build_superfunctional_from_dictionary(name, npoints, deriv, restricted):
+    check_consistency(dict_functionals[name], name)
+    if "xc_functionals" in dict_functionals[name].keys():
+        for xc_key in dict_functionals[name]["xc_functionals"].keys():
+            xc_name = "XC_" + xc_key
+        sup = core.SuperFunctional.XC_build(xc_name, restricted)
+        descr = "    " + name.upper() + " "
+        if sup.is_gga():
+            if sup.x_alpha() > 0:
+                descr += "Hyb-GGA "
+            else:
+                descr += "GGA "
+        descr += "Exchange-Correlation Functional\n"
+        sup.set_description(descr)
+    else:
+        sup = core.SuperFunctional.blank()
+        descr = []
+        citation = []
+        x_HF = {"ALPHA": 0.0, "OMEGA": 0.0, "BETA": 0.0}
+        if "x_functionals" in dict_functionals[name].keys():
+            x_funcs = dict_functionals[name]["x_functionals"]
+            for x_key in x_funcs.keys():
+                x_name = "XC_" + x_key
+                x_func = core.LibXCFunctional(x_name, restricted)
+                x_params = x_funcs[x_key]
+                if "use_libxc" in x_params and x_params["use_libxc"]:
+                    x_HF = x_func.query_libxc("XC_HYB_CAM_COEF")
+                    x_func.set_alpha(1.0)
+                if "tweak" in x_params.keys():
+                    x_func.set_tweak(x_params["tweak"])
+                if "alpha" in x_params.keys():
+                    x_func.set_alpha(x_params["alpha"])
+                if "omega" in x_params.keys():
+                    x_func.set_omega(x_params["omega"])
+                sup.add_x_functional(x_func)
+                if x_func.citation() not in citation:
+                    citation.append(x_func.citation())
+                if x_func.description() not in descr:
+                    descr.append(x_func.description())
+        if "x_hf" in dict_functionals[name].keys():
+            x_params = dict_functionals[name]["x_hf"]
+            if "use_libxc" in x_params.keys():
+                x_name = "XC_" + x_params["use_libxc"]
+                x_HF = core.LibXCFunctional(x_name, restricted).query_libxc("XC_HYB_CAM_COEF")
+            if "alpha" in x_params.keys():
+                sup.set_x_alpha(x_params["alpha"])
+            else:
+                x_params["alpha"] = 0.0
+            if "beta" in x_params.keys():
+                sup.set_x_beta(x_params["beta"])
+            if "omega" in x_params.keys():
+                sup.set_x_omega(x_params["omega"])
+        else:
+            sup.set_x_alpha(x_HF["ALPHA"] + x_HF["BETA"])
+            sup.set_x_beta(x_HF["ALPHA"] - x_HF["BETA"])
+            sup.set_x_omega(x_HF["OMEGA"])
+        if "c_functionals" in dict_functionals[name].keys():
```

Comment:
Can you add spaces between this `if` block and the previous along with a comment to describe what each ones does?